packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts

```diff
@@ -246,6 +246,155 @@ export const useNotifiSubscribe: ({
     setLoading,
   ]);
 
+  const updateAlertInternal = useCallback(
+    async (
+      alertParams: InstantSubscribe,
+      data: ClientData,
+      contacts: Readonly<{
+        finalEmail: string | null;
+        finalPhoneNumber: string | null;
+        finalTelegramId: string | null;
+      }>,
+    ): Promise<Alert | null> => {
+      const { alertName, alertConfiguration } = alertParams;
+      const { finalEmail, finalPhoneNumber, finalTelegramId } = contacts;
+      const existingAlert = data.alerts.find(
+        (alert) => alert.name === alertName,
+      );
+
+      const deleteThisAlert = async () => {
+        if (existingAlert !== undefined && existingAlert.id !== null) {
+          await client.deleteAlert({
+            alertId: existingAlert.id,
+            keepSourceGroup: keepSubscriptionData,
+            keepTargetGroup: keepSubscriptionData,
+          });
+        }
+      };
+
+      const ensureSource = async (
+        params: CreateSourceInput,
+      ): Promise<Source> => {
+        const existing = data.sources.find(
+          (s) =>
+            s.type === params.type &&
+            s.blockchainAddress === params.blockchainAddress,
+        );
+        if (existing !== undefined) {
+          return existing;
+        }
+
+        const created = await client.createSource(params);
+        return created;
+      };
+      if (alertConfiguration === null) {
+        await deleteThisAlert();
+        return null;
+      } else if (alertConfiguration.type === 'multiple') {
+        const {
+          filterOptions,
+          filterType,
+          sources: sourcesInput,
+        } = alertConfiguration;
+        const sources = await Promise.all(sourcesInput.map(ensureSource));
+        const filter = sources
+          .flatMap((s) => s.applicableFilters)
+          .find((f) => f.filterType === filterType);
+        if (filter === undefined || filter.id === null) {
+          await deleteThisAlert();
```

Comment:
is there a situation when we would need to delete the alert with a filter/source being undefined?

Reply:
ah I'm just keeping the old behavior for now I think.